stdlib/stdlib.mli

```diff
@@ -51,6 +51,17 @@ exception Exit
 (** The [Exit] exception is not raised by any library function.  It is
     provided for use in your programs. *)
 
+val try_finally : always:(unit -> unit) -> (unit -> 'a) -> 'a
+(** [try_finally ~always work] is designed to run code in [work] that
+    may fail with an exception.
+    The function [always], is guaranteed to run after {b any} execution
+    of the [work] function. Exceptions raised by [always] are not
+    caught, they will be propagated to the caller as usual.
+    In any other case [try_finally] will return the result of the
+    [work] funciton or re-raise its exception, preserving the
+    backtrace (For more details, see {!Printexc.raise_with_backtrace}).
```

Comment:
I think this doc string is unnecessarily long-winded. Here would be my take on it:

`try_finally ~always work`  invokes `work ()` and then `always ()` before `work` returns with its value or an exception. In the latter case the exception is re-raised after `always ()`,  preserving the backtrace (see `{!Printexc.raise_with_backtrace}`). If `always ()` raises, this exception is not caught and may shadow one `work ()` may have raised.

Or even shorter (I'm still convinced the discussion about the backtrace is pointless): 

`try_finally ~always work`  invokes `work ()` and then `always ()` before `work` returns with its value or an exception. In the latter case the exception is re-raised after `always ()`. If `always ()` raises, this exception is not caught and may shadow one `work ()` may have raised.

Reply:
I think that if we use the term "re-raise" explicitly, then it is reasonable to assume that this implicitly suggests "with the same backtrace" -- and indeed I hope that we can move in the future to a world where `reraise` preserves the backtrace always.